airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/write/object_storage/ObjectStorageStreamLoaderFactory.kt

```diff
@@ -65,60 +68,34 @@ class ObjectStorageStreamLoader<T : RemoteObject<*>, U : OutputStream>(
     private val pathFactory: ObjectStoragePathFactory,
     private val bufferedWriterFactory: BufferedFormattingWriterFactory<U>,
     private val destinationStateManager: DestinationStateManager<ObjectStorageDestinationState>,
-    private val partSize: Long,
+    private val recordBatchSizeBytes: Long,
 ) : StreamLoader {
     private val log = KotlinLogging.logger {}
 
-    sealed interface ObjectStorageBatch : Batch
-    data class RemoteObject<T>(
-        override val state: Batch.State = Batch.State.COMPLETE,
-        val remoteObject: T,
-        val partNumber: Long,
-        override val groupId: String? = null
-    ) : ObjectStorageBatch
-
-    private val partNumber = AtomicLong(0L)
+    // Used for naming files. Distinct from part index, which is used to track uploads.
+    private val fileNumber = AtomicLong(0L)
+    private val objectAccumulator = PartToObjectAccumulator(stream, client)
 
     override suspend fun start() {
         val state = destinationStateManager.getState(stream)
         val nextPartNumber = state.nextPartNumber
         log.info { "Got next part number from destination state: $nextPartNumber" }
-        partNumber.set(nextPartNumber)
+        fileNumber.set(nextPartNumber)
```

Comment:
If fileNumber is distinct from partNumber, why do we set the fileNumber equal to the nextPartNumber?

Reply:
The macro this populates is called "part number." I changed the class var to `fileNumber` to avoid confusion but didn't change the state interface. I will do that and add comments to avoid confusion.